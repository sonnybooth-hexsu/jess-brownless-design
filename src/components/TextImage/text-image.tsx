import React, { ReactNode } from "react"
import {
  textImageContainer,
  textImageChildren,
  textImageChildrenReverse,
  textImageContainerReverse,
} from "./text-image.module.css"

type TextImageProps = {
  image: string
  imageAlt: string
  reverse?: boolean
  children: ReactNode
  additionalClasses: string
}

export const TextImage = ({
  image,
  imageAlt,
  reverse,
  children,
}: TextImageProps) => {
  return (
    <div className={`${reverse ? textImageContainerReverse : ""}`}>
      {reverse ? (
        <div className={`container mx-auto ${textImageContainer}`}>
          <img
            loading="lazy"
            data-testid="textImageCoverReverse"
            src={image}
            alt={imageAlt}
          />
          <div className={textImageChildrenReverse}>{children}</div>
        </div>
      ) : (
        <div className={textImageContainer}>
          <img
            loading="lazy"
            data-testid="textImageCover"
            src={image}
            alt={imageAlt}
          />
          <div className={textImageChildren}>{children}</div>
        </div>
      )}
    </div>
  )
}
