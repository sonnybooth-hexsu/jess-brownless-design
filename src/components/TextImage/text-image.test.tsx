/* eslint no-undef: 0 */

import React from "react"
import { render } from "@testing-library/react"
import { getByText, getByTestId, getByAltText } from "@testing-library/dom"
import { TextImage } from "./text-image"
import cover1 from "../assets/images/cover-1.jpg"

describe("<TextImage />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <TextImage image={cover1} imageAlt="Girl smiling" additionalClasses="">
        <h2 data-testid="textImageChild">I am test child</h2>
      </TextImage>
    )

    expect(container).toMatchSnapshot()
  })

  it("should render child", () => {
    const { container } = render(
      <TextImage image={cover1} imageAlt="Girl smiling" additionalClasses="">
        <h2 data-testid="textImageChild">I am test child</h2>
      </TextImage>
    )

    const childEl = getByTestId(container, "textImageChild")

    expect(getByText(childEl, "I am test child"))
  })

  it("should render image alt", () => {
    const { container } = render(
      <TextImage image={cover1} imageAlt="Girl smiling" additionalClasses="">
        <h2 data-testid="textImageChild">I am test child</h2>
      </TextImage>
    )

    expect(getByAltText(container, "Girl smiling"))
  })

  it("reverses image and text", () => {
    const { container } = render(
      <TextImage
        reverse
        image={cover1}
        imageAlt="Girl smiling"
        additionalClasses=""
      >
        <h2 data-testid="textImageChild">I am test child</h2>
      </TextImage>
    )

    expect(getByTestId(container, "textImageCoverReverse"))
  })
})
