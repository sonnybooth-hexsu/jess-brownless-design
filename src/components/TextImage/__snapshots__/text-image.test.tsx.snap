// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextImage /> should match snapshot 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="textImageContainer"
    >
      <img
        alt="Girl smiling"
        data-testid="textImageCover"
        loading="lazy"
        src="test-file-stub"
      />
      <div
        class="textImageChildren"
      >
        <h2
          data-testid="textImageChild"
        >
          I am test child
        </h2>
      </div>
    </div>
  </div>
</div>
`;
